### ISSUE TITLE
Fading shows desktop

### ISSUE THREAD

**Issue by Zekario** [chjj/compton#342]
Using i3, switching workspaces causes a nice fadein/out from compton, however, it shows the desktop background, which can get annoying. Is there a way to make compton use a solid color for a background when fading, or not show the background at all?

**Comment by krompus** [comment]
I have the same setup and issue; it would be nice to know if there's a fix. I originally thought it was i3's fault, since I've been using compton with it for so long, but when I disabled compton, the background flash went away. I'll keep searching. :)

**Comment by Chilinot** [comment]
I have the same issue

**Comment by cju** [comment]
… confirmed…

**Comment by richardgv** [comment]
Well, when a window is being faded out, you see whatever is under it, and in the case of i3, it is your wallpaper (or more technically, you see the contents of the pixmap specified in the `_XROOTPMAP_ID` or `_XSETROOT_ID` property of the root window, or gray if none of the properties exists). I do not see what is wrong with the behavior. If you do not like the appearance, simply choose a wallpaper you like: If you like solid color, use a wallpaper with solid color. I do not see where the problem is. Did I understand your requirements incorrectly?

**Comment by onodera-punpun** [comment]
This could be fixed with a suggestion/issue someone reported here earlier as well: https://github.com/chjj/compton/issues/314

**Comment by Chilinot** [comment]
Personally i would like to be able to disable the fading. The fading makes the workspace switching less rapid and slow.

**Comment by richardgv** [comment]
@onodera-punpun and @Chilinot:

Okay, I will look into #314 after I take care of the bugs, but I am afraid it would not be too soon.

Meanwhile, you could try `--no-fading-openclose`, which does disable fading during workspace switches for i3 -- but it disables fading when opening/closing windows, too.

**Comment by Chilinot** [comment]
@richardgv thanks, that fixed it :) personally i can live without the open/close fading :)

**Comment by JunnanZ** [comment]
Hi guys, I had the same issue and I just came up with a better solution. The reason we can see the wallpaper when fading is that the previous screen is fading out too fast, or the latter screen fading in too slow. Then, at some point, the sum of two screens' opacity drops below 1 and we see the wallpaper. So what we should do is adjusting the fade-in-step and fade-out-step to prevent this from happending. 

Try this: `compton -D 5 -I 0.05 -O 0.02`. It prevents the wallpaper from showing up while preserves the beautiful fading effect. Fine-tune these values to suit your preference but keep fade-in-step slightly larger than fade-out-step. Hope this helps!

**Comment by Champitoad** [comment]
I still get to see my wallpaper for a fraction of second when switching workspace and with the `--no-fading-openclose` option.

**Comment by BlackCapCoder** [comment]
I am having this issue in xmonad, with or without `--no-fading-openclose`

**Comment by rifazn** [comment]
@BlackCapCoder Try @JunnanZ's suggestion but tweak the fade-in (`-I`) and fade-out (`-O`) values if it doesn't fix it for you. In my case I basically copy pasted his code but decreased the `-O` to 0.01 and its working now.

For your convenience, this is what I'm running now:
`compton -D 5 -I 0.03 -O 0.01 -f`

Note: the `-f` flag is for turning on fade-in/fade-out, it wasn't mentioned in @JunnanZ suggestion.

**Comment by joelostblom** [comment]
I agree that it would be useful with an option to "Fade via black" (or a customizable color) instead of via the desktop background. Fading via black gives the appearance that the windows morph/fade into the windows of the next workspace, rather than seeing the wallpaper flicker by for a split second.

They fading looks great with a black wallpaper, so I might just switch out my wallpaper (and put it as the lockscreen instead since I like the image).

**Comment by rudevdr** [comment]
If windows are full screen the background is not shown.

Also see [this](https://www.reddit.com/r/i3wm/comments/40h1q9/comption_fading_between_workspaces_shows_desktop/cyu4pit/)
> Switching a workspace is done by unmapping all windows on the source workspace and mapping the windows of the target workspace. In other words, it's just making certain windows disappear and others appear. From a X point of view, this is no different from an application starting up and opening a window.

**Comment by jordiorlando** [comment]
@rudevdr this still happens for me even if both windows are fullscreen

**Comment by ghost** [comment]
A solution to lessen this is to make your fade out step less than your fade in step. You may have to mess around with the values until you find the speed you prefer.

**Comment by tjbrn** [comment]
This problem still bothering me. Any solution?

**Comment by rigopoui** [comment]
I think a solution would be to change the way i3 handles workspace changes, if possible. As @rudevdr mentioned:

> Switching a workspace is done by unmapping all windows on the source workspace and mapping the windows of the target workspace. In other words, it's just making certain windows disappear and others appear. From a X point of view, this is no different from an application starting up and opening a window.

If i3 were to first render the new windows on top of the old ones, and then close the old ones, then the desktop would not be visible during the transition.
